tests/Facebook/InstantArticles/Transformer/CMS/wp-ia.xml

```diff
@@ -17,6 +17,20 @@
       </header>
       <p>Yes, peace is good for everybody!<br/>
 Man kind.</p>
+      <p>Some text comes here and some more and more.<a href="http://example.com/image1.jpg">...</a></p>
+      <figure>
+        <img src="http://example.com/image1.jpg"/>
+      </figure>
+      <p>Some text comes here and some more and more.<a href="http://example.com/image2.jpg">...</a> and some.</p>
```

Comment:
That doesn't look like the expected behavior. If the image is in the middle of the text we should close the Paragraph, add the image and open another one.

What if the image is the first thing in the Paragraph? Doesn't it make more sense to add it before the Paragraph instead of after?

This is hard as there is no concept of `appendBefore` on the SDK, and by the time you run this rule, you had already added the Paragraph to root.

My take on this would be:

1/ Do not do this for any TextContainer, but only root Paragraphs, so we know how to close/open them.
2/ The Rule should return a newly created (and appended) Paragraph as a context for the Transformer to continue processing the document.